Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -768,16 +767,23 @@ class BrowserViewController: UIViewController {
         })
     }
 
-    fileprivate func updateInContentHomePanel(_ url: URL?) {
+    fileprivate func updateInContentHomePanel(_ url: URL?, didSelectTab: Bool) {
         let isAboutHomeURL = url.flatMap { InternalURL($0)?.isAboutHomeURL } ?? false
         if !urlBar.inOverlayMode {
             guard let url = url else {
+                navigationToolbar.updateIsSearchStatus(true)
                 hideFirefoxHome()
                 return
             }
             if isAboutHomeURL {
                 showFirefoxHome(inline: true)
             } else if !url.absoluteString.hasPrefix("\(InternalURL.baseUrl)/\(SessionRestoreHandler.path)") {
+                // Fix: During transition from tabTray to tab view   the progress bar doesn't reflect correct state and hence we ignore changing any state of the middle search button 
+                if !didSelectTab {
```

Comment:
I am not understanding the `didSelectTab` requirement, IIUC this PR is connecting the progress bar state to the state of the Search/Reload button.